tests/FSharp.Compiler.ComponentTests/Language/StateMachineTests.fs

```diff
@@ -160,4 +160,17 @@ let main _ =
         |> ignoreWarnings
         |> withOptimize
         |> compileExeAndRun
-        |> shouldSucceed
\ No newline at end of file
+        |> shouldSucceed
+
+    [<Fact>]
+    let ``State machine defined as top level value is statically compiled`` () =
```

Comment:
Can you add an IL test so that we can see what happens with the placement for top-level state machine bindings?